app/js/components/service_status.ts

```diff
@@ -92,6 +86,56 @@ export class ServiceStatus {
       tooltips.push(states.service.tooltips[key]);
     }
 
+    const customTooltip = function (this: any, tooltipModel: any) {
```

Comment:
Could the tooltip part be levered to an individual file? Making the service_status script more OO.

Reply:
I do like if this is in the same file as the component. I agree it's not very good quality code but I don't like putting it far away either.